crates/rome_js_parser/src/syntax/typescript/types.rs

```diff
@@ -640,7 +675,7 @@ fn parse_ts_mapped_type(p: &mut JsParser) -> ParsedSyntax {
     p.expect(T!['[']);
     parse_ts_type_parameter_name(p).or_add_diagnostic(p, expected_ts_type_parameter);
     p.expect(T![in]);
-    parse_ts_type(p).or_add_diagnostic(p, expected_ts_type);
+    parse_ts_type(p, TypeContext::default()).or_add_diagnostic(p, expected_ts_type);
```

Comment:
`TypeContext::default` should only used in positions that start a new "type-parsing" context (e.g. because it is parenthesized or because it is in an initializer position). It's otherwise best to pass the context to all `parse_ts_type` functions

```suggestion
    parse_ts_type(p, context).or_add_diagnostic(p, expected_ts_type);
```

The same applies to many other positions (e.g. mapped types, tuple types etc)

Reply:
@MichaReiser 
> The only thing that we need to be careful about is to make sure to pass context through correctly and only use TypeContext::default in positions where a type starts a complete new context (because it is parenthesized)

It is really hard for me to pass context through correctly. I tried, but I couldn't do that. There are too many functions which the context should be passed to. I would like you to do  this refactoring on your end if possible. If you don't, I want you to merge # and create another PR for this refactoring.